experimental/experimental.go

```diff
@@ -28,23 +28,24 @@ package experimental
 import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/internal"
+	"google.golang.org/grpc/mem"
 )
 
 // WithRecvBufferPool returns a grpc.DialOption that configures the use of
 // bufferPool for parsing incoming messages on a grpc.ClientConn. Depending on
 // the application's workload, this could result in reduced memory allocation.
 //
 // If you are unsure about how to implement a memory pool but want to utilize
-// one, begin with grpc.NewSharedBufferPool.
+// one, begin with mem.NewBufferPool.
 //
 // Note: The shared buffer pool feature will not be active if any of the
 // following options are used: WithStatsHandler, EnableTracing, or binary
 // logging. In such cases, the shared buffer pool will be ignored.
 //
 // Note: It is not recommended to use the shared buffer pool when compression is
 // enabled.
-func WithRecvBufferPool(bufferPool grpc.SharedBufferPool) grpc.DialOption {
-	return internal.WithRecvBufferPool.(func(grpc.SharedBufferPool) grpc.DialOption)(bufferPool)
+func WithRecvBufferPool(bufferPool mem.BufferPool) grpc.DialOption {
```

Comment:
Should we rename this to `WithBufferPool` since this buffer pool is now used for the send and receive sides?

Reply:
Good idea, that makes more sense